tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ListingsSpec.scala

```diff
@@ -382,4 +388,87 @@ final class ListingsSpec extends BaseIntegrationSpec {
 
   }
 
+  "Fulltext search" should {
+
+    val project = ref11
+
+    "find a match by @id" in {
+      val id  = s"http://bbp.epfl.ch/${genString()}"
+      val id2 = s"http://bbp.epfl.ch/${genString()}"
+
+      val resource: String => Json = id => json"""{ "@id": "$id" }"""
+
+      postResource(resource(id), project).accepted
+      postResource(resource(id2), project).accepted
+
+      eventually {
+        fulltextListing(q = UrlUtils.encode(id), project) { json =>
+          val results = listing._results.getOption(json).value
+          results.loneElement should have(`@id`(id))
+        }
+      }
+    }
+
+    "find the match on @id first even when other docs reference it" in {
+      val id = s"http://bbp.epfl.ch/${genString()}"
+
+      val resource: String => Json = id => json"""{ "@id": "$id" }"""
+      val resRef                   = json"""{ "http://schema.org/description": "$id" }"""
+      val resRef2                  = json"""{ "randomField": "$id" }"""
+
+      postResource(resource(id), project).accepted
+      postResource(resRef, project).accepted
+      postResource(resRef2, project).accepted
+
+      eventually {
+        fulltextListing(q = UrlUtils.encode(id), project) { json =>
+          val results = listing._results.getOption(json).value
+          results.head should have(`@id`(id))
+        }
+      }
+    }
+
+    "find match by name" in {
+      val id       = s"http://bbp.epfl.ch/${genString()}"
+      val resource = json"""{ "@id": "$id", "http://schema.org/name": "Lorem ipsum dolor"}"""
+
+      postResource(resource, project).accepted
+
+      val query = UrlUtils.encode("lor ip dol")
+
+      eventually {
+        fulltextListing(q = query, project) { json =>
+          val results = listing._results.getOption(json).value
+          results.head should have(`@id`(id))
+        }
+      }
+    }
+
+    "find match by description" in {
+      val id       = s"http://bbp.epfl.ch/${genString()}"
+      val resource =
+        json"""{ "@id": "$id", "http://schema.org/description": "Northumberland is a ceremonial county bordering Scotland."}"""
```

Comment:
For understanding, why do name and description need their full context?

Reply:
This resource has no context, so if we put `"description"` then it's not going to be `http://schema.org/description` but instead it's going to be using the vocab defined in the project. If the non schema.org is pushed to ES then the fields is not `"description"` and the mapping does not apply to it.

Usually, in prod projects this is not specified in each resource, instead the context is defined in a separate resource and just referenced by `@id`.